catalog/lib/widgets/app_selection_control.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:catalog/catalog.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class AppSelectionControl<T> extends StatelessWidget {
+  final bool isSelected;
+  final String label;
+  final VoidCallback onPressed;
+  final bool? isRadioButton;
+
+  const AppSelectionControl({
+    required this.isSelected,
+    required this.label,
+    required this.onPressed,
+    this.isRadioButton = true,
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) => SizedBox(
+        height: 40.h,
+        child: Row(
+          children: [
+            Expanded(
+              child: IconButton(
+                padding: EdgeInsets.symmetric(vertical: 0.h),
+                onPressed: onPressed,
+                icon: Row(
+                  children: [
+                    isSelected
```

Comment:
What do you think about extracting this as a function that returns the proper Icon widget? 
```suggestion
                    _getIcon()
```